pkg/v1/tkg/test/tkgctl/shared/package.go

```diff
@@ -0,0 +1,125 @@
+
+	"github.com/vmware-tanzu/tanzu-framework/pkg/v1/tkg/constants"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	clusterapiv1beta1 "sigs.k8s.io/cluster-api/api/v1beta1"
+
+	kappctrl "github.com/vmware-tanzu/carvel-kapp-controller/pkg/apis/kappctrl/v1alpha1"
+	kapppkgiv1alpha1 "github.com/vmware-tanzu/carvel-kapp-controller/pkg/apis/packaging/v1alpha1"
+	addonutil "github.com/vmware-tanzu/tanzu-framework/addons/pkg/util"
+	runtanzuv1alpha3 "github.com/vmware-tanzu/tanzu-framework/apis/run/v1alpha3"
+)
+
+const (
+	waitTimeout     = time.Second * 90
+	pollingInterval = time.Second * 2
+)
+
+// getClusterBootstrap gets ClusterBootstrap resource with the provided object key
+func getClusterBootstrap(ctx context.Context, k8sClient client.Client, namespace, clusterName string) *runtanzuv1alpha3.ClusterBootstrap {
+	clusterBootstrap := &runtanzuv1alpha3.ClusterBootstrap{}
+	objKey := client.ObjectKey{Namespace: namespace, Name: clusterName}
+
+	Eventually(func() bool {
+		err := k8sClient.Get(ctx, objKey, clusterBootstrap)
+		return err == nil
+	}, waitTimeout, pollingInterval).Should(BeTrue())
+
+	Expect(clusterBootstrap).ShouldNot(BeNil())
+	return clusterBootstrap
+}
+
+// getPackageInstall get PackageInstall resource with the provided object key
+func getPackageInstall(ctx context.Context, k8sClient client.Client, namespace, pkgiName string) *kapppkgiv1alpha1.PackageInstall {
+	pkgInstall := &kapppkgiv1alpha1.PackageInstall{}
+	objKey := client.ObjectKey{Namespace: namespace, Name: pkgiName}
+
+	Eventually(func() bool {
+		err := k8sClient.Get(ctx, objKey, pkgInstall)
+		return err == nil
+	}, waitTimeout, pollingInterval).Should(BeTrue())
+
+	Expect(pkgInstall).ShouldNot(BeNil())
+
+	return pkgInstall
+}
+
+func getPackageDetailsFromCBS(CBSRefName string) (string, string, string, error) {
+	pkgShortName := strings.Split(CBSRefName, ".")[0]
+
+	pkgName := strings.Join(strings.Split(CBSRefName, ".")[0:4], ".")
+
+	pkgVersion := strings.Join(strings.Split(CBSRefName, ".")[4:], ".")
+
+	return pkgShortName, pkgName, pkgVersion, nil
+}
+
+func checkPackageInstalls(ctx context.Context, cl client.Client, scheme *runtime.Scheme, testPkgName string) error {
+	var (
+		err          error
+		pkgShortName string
+		pkgName      string
+		pkgVersion   string
+	)
+
+	mngCluster := &clusterapiv1beta1.Cluster{}
+
+	// create remoteClient for workload cluster
+	remoteClient, err := addonutil.GetClusterClient(ctx, cl, scheme, util.ObjectKey(mngCluster))
+	if err != nil {
+		return err
+	}
+
+	// Get ClusterBootstrap and return error if not found
+	clusterBootstrap := getClusterBootstrap(ctx, cl, constants.TkgNamespace, mngCluster.Name)
+
+	// packageInstall name for for both management and workload clusters should follow the <cluster name>-<addon short name>
+	// packageInstall name and version should match info in clusterBootstrap for all packages, format is <package name>.<package version>
+	switch {
+	case testPkgName == "CNI":
```

Comment:
This function doesnt need to take in testPkgName as input. Instead can just get package name and versions from CB and verify if they are successful.

i.e.

```
cnipkgShortName, cnipkgName, cnipkgVersion, err = getPackageDetailsFromCBS(clusterBootstrap.Spec.CNI.RefName)

csipkgShortName, csipkgName, csipkgVersion, err = getPackageDetailsFromCBS(clusterBootstrap.Spec.CSI.RefName)

verifyPackageInstall(cnipkgShortName, cnipkgName, cnipkgVersion)
verifyPackageInstall(csipkgShortName, csipkgName, csipkgVersion)

func verifyPackageInstall() {

pkgiName := addonutil.GeneratePackageInstallName(mngCluster.Name, pkgShortName)
	pkgi := getPackageInstall(ctx, remoteClient, constants.TkgNamespace, pkgiName)

	// check package install reconcile status is succeed
	Expect(pkgi.Status.GenericStatus.Conditions[1].Type).Should(Equal(kappctrl.ReconcileSucceeded))
	Expect(pkgi.Status.GenericStatus.Conditions[1].Status).Should(Equal(corev1.ConditionTrue))

	// Verify package name match between clusterBootstrap and packageInstall
	Expect(pkgName).Should(Equal(pkgi.Spec.PackageRef.RefName))

	// Verify package version match between clusterBootstrap and packageInstall
	Expect(pkgVersion).Should(Equal(pkgi.Spec.PackageRef.VersionSelection.Constraints))
}
```

Reply:
modified with new structure and don't need to input package name, will check all packages at once